game-app/game-core/src/main/java/games/strategy/triplea/ui/menubar/DebugMenu.java

```diff
@@ -26,9 +25,7 @@
 public final class DebugMenu extends JMenu {
   private static final long serialVersionUID = -4876915214715298132L;
 
-  /** Maps the debug menu title to the function that will create the sub menu items */
-  private static final Map<String, Function<TripleAFrame, Collection<JMenuItem>>>
-      menuItemsAndFactories = new TreeMap<>();
+  private static final List<Consumer<TripleAFrame>> frameFactories = new ArrayList<>();
```

Comment:
Consider renaming this variable from `frameFactories` to `frameVisitors`. `factories` really caught my attention as a name - I think I realized it is because we are not really using the factory pattern here but instead are using the 'Visitor' pattern (or at least, I think we are, looks like visitor :smile: :wave: :person_fencing: )

Reply:
Thanks, I also had a problem with that, but didn't want to change the name as I wasn't sure why it was chose in the first place.
It's corrected now.